### PR TITLE
feat (scipy.special): Add a xla version of scipy.special.gamma function

### LINKED ISSUE #15409: AttributeError: module 'jax.scipy' has no attribute 'gamma'
Hi everyone, I was wondering why the gamma() function is missing, while gammaln() is present.
I suggest you add it in a future edit, because this raises an AttributeError using the function pytensor.tensor.math.gamma() in a pymc model. Thank you so much for your work.

### PR DESCRIPTION
**Pull request by laqua-stack**
- Add low-level api in lax.special
 - Add high-level api in scipy.special
 - Add tests for this purpose

Currently, there is no implementation of the gamma function in jax
 but there is one in scipy.special. This breaks some higher level
  jit-compilation like in the blackjax backend for pymc. This commit
  adds the missing gamma function.

Resolves: #15409

### CODE REVIEW

File: jax/_src/lax/special.py [hunk 1]
```diff
@@ -45,6 +45,10 @@ def lgamma(x: ArrayLike) -> Array:
   r"""Elementwise log gamma: :math:`\mathrm{log}(\Gamma(x))`."""
   return lgamma_p.bind(x)
 
+def gamma(x: ArrayLike) -> Array:
+  r"""Elementwise gamma: :math:`\Gamma(x)`."""
+  return exp(lgamma_p.bind(x))
```
Comment (jakevdp) [inline jax/_src/lax/special.py:50]: I would probably avoid adding any new function here, unless it's a nontrivial implementation.

**Review by jakevdp — CHANGES_REQUESTED** [review]
Looks great!

Please also add the new function to the list here: https://github.com/google/jax/blob/main/docs/jax.scipy.rst#jaxscipyspecial

And when you're finished, please squash the changes into a single commit (see https://jax.readthedocs.io/en/latest/contributing.html#single-change-commits-and-pull-requests)

Thanks!

File: jax/_src/lax/special.py (hunk 1 above)
Comment (laqua-stack) [reply jax/_src/lax/special.py:50]: I am sorry, I did not see this before adding docs... I think I need to reopen the PR and fix it

**Comment by laqua-stack** [comment]
@jakevdp Could you maybe reopen the PR? Somehow I managed to get it automatically closed :-(

**Comment by jakevdp** [comment]
OK - for what its worth the PR was automatically closed because you deleted the branch that it was based on. In the future, you could push the branch to github again and re-open the same PR.

**Comment by laqua-stack** [comment]
Thank you!

**Pull request closed without merging**